task-driver/integration/helpers.rs

```diff
@@ -125,36 +147,110 @@ pub async fn mock_wallet_update(wallet: &mut Wallet, client: &ArbitrumClient) ->
 
     // Mock a `VALID WALLET UPDATE` proof bundle
     let mut proof = dummy_valid_wallet_update_bundle();
+    proof.statement.external_transfer = ExternalTransfer::default();
     proof.statement.old_shares_nullifier = nullifier;
     proof.statement.new_private_shares_commitment = share_comm;
     proof.statement.new_public_shares = wallet.blinded_public_shares.clone();
 
     client.update_wallet(proof, vec![] /* statement_sig */).await.map_err(Into::into)
 }
 
+/// Fund the wallet in the integration test args with a given amount of WETH and
+/// the dummy erc20 token
+pub(crate) async fn fund_wallet(test_args: &IntegrationTestArgs) -> Result<()> {
+    let weth_client =
+        create_weth_client(test_args).map_err(|e| eyre!("error creating weth client: {e}"))?;
+    let erc20_client = create_erc20_client(&test_args.erc20_addr, test_args)
+        .map_err(|e| eyre!("error creating erc20 client: {e}"))?;
+
+    let addr = test_args.arbitrum_client.wallet_address();
+
+    // First wrap native ETH to fund WETH
+    let weth_balance =
+        weth_client.balance_of(addr).await.map_err(|e| eyre!("error getting weth balance: {e}"))?;
+    let funds_needed = FUNDING_AMOUNT.saturating_sub(weth_balance.as_u64());
+    if funds_needed > 0 {
+        wrap_eth(funds_needed, test_args).await?;
+    }
+
+    // Now mint erc20 tokens to fund the wallet
+    let erc20_balance = erc20_client
+        .balance_of(addr)
+        .await
+        .map_err(|e| eyre!("error getting erc20 balance: {e}"))?;
+
+    let funds_needed = FUNDING_AMOUNT.saturating_sub(erc20_balance.as_u64());
+    if funds_needed > 0 {
+        fund_wallet_with_erc20(funds_needed, test_args).await?;
+    }
+
+    Ok(())
```

Comment:
thoughts on also approving the darkpool to spend the wallet's funds here before returning? seems like a sensible place to do it